catalog/spatial/geocoding/spatial-geocoding-api/src/main/java/org/codice/ddf/spatial/geocoding/GeoNamesRemoteDownloader.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+
+package org.codice.ddf.spatial.geocoding;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class GeoNamesRemoteDownloader  {
+
+    private static final int BUFFER_SIZE = 4096;
+
+    private String indexLocation = System.getProperty("user.dir") + "/data/tmp";
+
+    /**
+     * Used for testing purposes.
+     *
+     * @param indexLocation
+     */
+
+    public void setIndexLocation(String indexLocation) {
+        this.indexLocation = indexLocation;
+    }
+
+    /**
+     * Download a GeoNames .zip file from a remote location
+     *
+     * @param url - the URL of the GeoNames zip file ( ex: http://download.geonames.org/export/dump/AD.zip )
+     *
+     * @param progressCallback -  the callback to receive updates about the indexing progress, may be
+     *                          null if you don't want any updates
+     */
+    public String getResourceFromUrl(String url, final ProgressCallback progressCallback)
```

Comment:
I think you should consider refactoring this method into 2 utility methods.  One would return an InputStream given an HTTP url String.  The other would accept an input stream and write those contents to a file.  That utility should be accessible by other bundles.